src/ocean/util/app/ext/ReopenableFilesExt.d

```diff
@@ -70,14 +94,54 @@ public class ReopenableFilesExt : IApplicationExtension, ISignalExtExtension
 
     ***************************************************************************/
 
-    public this ( SignalExt signal_ext = null, int reopen_signal = SIGHUP )
+    public this ( SignalExt signal_ext = null,
+        int reopen_signal = SIGHUP )
     {
+        this(signal_ext, null, reopen_signal);
+    }
+
+
+    /***************************************************************************
+
+        Constructor. Optionally registers this extension with the signal
+        extension and/or unix socket extension, activates the handling of the
+        specified signal, which will cause the registered files to be reopened,
+        and activates the handling of the specified unix socket command, which
+        will cause the registered file to be reopened.
+
+        Params:
+            signal_ext = optional SignalExt instance to register with
+            unix_socket_ext = optional UnixSocketExt instance to register with
+            reopen_signal = if signal_ext is non-null, signal to trigger
+                reopening of registered files
+            reopen_command = if unix socket ext is non-null, command to trigger
+                reopenning of the registered file (passed as the first
+                argument to the command).
+
+    ***************************************************************************/
+
+    public this ( SignalExt signal_ext,
+        UnixSocketExt unix_socket_ext,
+        int reopen_signal = SIGHUP,
+        cstring reopen_command = "reopen_logfile")
```

Comment:
Do we really want to make this configurable? It seems strange to be able to introduce inconsistencies in the socket interface...

Reply:
Well, we do the same thing for the SIGHUP. Note that people don't actually use this class directly, but it's used withing DaemonApp.

Reply:
Travis says this should be an `istring`
Also this works with *any* file opened with `ReusableFilesExt` right ? In this case `logfile` might be a bit too restrictive of a name.

Reply:
Another comment about this: It looks weird to have signal-related thing, unixsocket, signal-related, unix-socket.

I understand why you went with this, and the alternative is not too appealing as well, so I'm going to propose an alternate solution: since both `SignalExt` and `UnixSocketExt` are optional and independent of each other, why don't we move the logic to `setup` methods, e.g. `setupUnixSocket(UnixSocketExt, istring = "reopen_file")` and `setupSignalExt(SignalExt, int = SIGHUP)` (naming up for discussion).

Reply:
Right, deprecating constructor and having setup methods replace it would work, good idea :+1:

Reply:
Damn, I can't deprecate the existing constructor as all the methods are optional, so simply `auto ext = new ReopenableFilesExt` will trigger the deprecation :(.